### PR TITLE
Matchcommand/05 added tostring (#43)

### PR DESCRIPTION
**Pull request by cerulyean**
Adds a toString() method to MatchCommand using ToStringBuilder, consistent with other command classes.

The method includes key matching attributes:isBuyer, district, landSize, price

This improves debuggability and ensures meaningful output when commands are logged or printed during testing.

### CODE REVIEW

**Review by jhwan0707 — COMMENTED** [review]
The MatchCommand implementation looks solid, but there are areas where test coverage needs improvement. Below are my comments on the changes:

Code Coverage
Several lines (L29-31, L35, L37-38, L40-41, etc.) are not covered by tests. Consider adding unit tests to ensure all execution paths are tested, especially edge cases like null values and boundary conditions.

Code Quality & Maintainability
getMatchPredicate() (L45-72): This method has multiple if conditions that could be extracted into helper methods to improve readability.

equals() method (L82-88): Consider using Objects.equals() for null safety and cleaner comparison.

toString() method (L97-102): No major issues, but ensure it aligns with how other command classes implement toString().

Git Standards
The PR title could be more descriptive. Something like "Improve MatchCommand test coverage and refactor predicate logic" would be clearer.

Ensure commit messages follow a structured format (e.g., "MatchCommand: Add tests for getMatchPredicate()").

Overall, the PR is well-structured, but adding test cases and minor refactoring would improve maintainability. Let me know if you need any help with test cases!

File: src/main/java/seedu/address/logic/commands/MatchCommand.java [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+        model.updateFilteredPersonList(matchPredicate);
+
+        int matchCount = model.getFilteredPersonList().size();
+        return new CommandResult(String.format(Messages.MESSAGE_PERSONS_MATCH_OVERVIEW, matchCount));
+    }
+
+    private Predicate<Person> getMatchPredicate(Person person) {
+        return candidate -> {
+            if (person.getIsBuyer() == null || candidate.getIsBuyer() == null) {
+                return false;
```
Comment (jhwan0707) [inline src/main/java/seedu/address/logic/commands/MatchCommand.java:47]: Would it be better to throw an exception instead of silently returning false? If this case should never happen, handling it explicitly could prevent silent failures.

File: src/main/java/seedu/address/logic/commands/MatchCommand.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+            }
+
+            if (person.getDistrict() == null
```
Comment (jhwan0707) [inline src/main/java/seedu/address/logic/commands/MatchCommand.java:50]: This block repeats null checks. Consider extracting it into a helper method like hasValidAttributes(person, candidate) for better readability.

File: src/main/java/seedu/address/logic/commands/MatchCommand.java [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+                    || candidate.getDistrict() == null
+                    || person.getLandSize() == null
+                    || candidate.getLandSize() == null
+                    || person.getPrice() == null
+                    || candidate.getPrice() == null) {
+                return false;
+            }
+
+            if (person.getIsBuyer()) {
```
Comment (jhwan0707) [inline src/main/java/seedu/address/logic/commands/MatchCommand.java:59]: The nested if structure could be simplified for readability. Perhaps use an early return:

if (person.getIsBuyer() == candidate.getIsBuyer()) {
    return false;
}

File: src/main/java/seedu/address/logic/commands/MatchCommand.java [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    public boolean equals(Object other) {
+        if (other == this) {
+            return true;
+        }
+        if (!(other instanceof MatchCommand)) {
+            return false;
+        }
+
+        MatchCommand otherMatch = (MatchCommand) other;
+        return person.getIsBuyer().equals(otherMatch.person.getIsBuyer())
```
Comment (jhwan0707) [inline src/main/java/seedu/address/logic/commands/MatchCommand.java:89]: Consider using Objects.equals() for null safety, e.g.:
`return Objects.equals(person.getIsBuyer(), otherMatch.person.getIsBuyer())
        && Objects.equals(person.getDistrict(), otherMatch.person.getDistrict())
        && Objects.equals(person.getLandSize(), otherMatch.person.getLandSize())
        && Objects.equals(person.getPrice(), otherMatch.person.getPrice());`
This avoids potential NullPointerException when comparing fields.

**Review by jhwan0707 — COMMENTED** [review]
The logic is well-structured, but small improvements in readability, error handling, and test coverage would make it even more robust. Let me know if you'd like me to help with test cases!
